[Myron Neo] iP

meerian opened the pull request:
# Duke 

>_"A failure to plan is a plan to fail."_ - Wise Man

This is a project named after the Java mascot _Duke_. Given below are instructions on how to use it.
Duke helps you become a _much better_ person, __never__ again missing your deadlines!
## Features
   ```
   Hello from
    ____        _        
   |  _ \ _   _| | _____ 
   | | | | | | | |/ / _ \
   | |_| | |_| |   <  __/
   |____/ \__,_|_|\_\___|
   ```
Features include:
* Adding various tasks such as deadlines and events
* Marking tasks that are completed
* Listing out all your tasks in a nice concise manner
* `Search` function
* Save/Load feature for your task list

How to use:
1. Download any of the releases from https://github.com/meerian/ip/releases
2. Run the .jar file
3. ???
4. Profit! 😎

Task List:
- [x] Done
- [ ] Not Done

Here's a sample block code:
```javascript
function fancyAlert(arg) {
  if(arg) {
    $.facebox({div:'#foo'})
  }
}
```

Commit:
Add Gradle support

meerian committed:
Level 1. Greet, Echo, Exit completed.

meerian committed:
Level 2: Add, List completed.

meerian committed:
Level 3. Mark as Done

meerian committed:
Level 4: ToDos, Events, Deadlines

meerian committed:
Refactored code to improve legibility.

meerian committed:
Refactored code to improve legibility.

meerian committed:
Level 5. Handle Errors completed.

meerian committed:
Level 6. Delete completed.

meerian committed:
Improved abstraction.

meerian committed:
Tried out A-TextUiTesting with runtest.bat

meerian committed:
Done Level 7. Save

meerian committed:
Rename Checker.java to Parser.java for better clarity

meerian committed:
Teach Duke how to understand dates and times.

meerian committed:
Merge remote-tracking branch 'origin/branch-Level-7'

# Conflicts:
#	src/main/java/Parser.java

meerian committed:
Revert "Merge remote-tracking branch 'origin/branch-Level-7'"

This reverts commit 68531ab0498b5516ea6a28e56125b50748f1e136, reversing
changes made to fb4a41fc7b23fb740fc2032f96746d7352d41184.

Revert "Merge remote-tracking branch 'origin/branch-Level-7'"

meerian committed:
Merge Level-7 branch into main branch

meerian committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Parser.java

meerian committed:
Fix code for Parser.java

meerian committed:
Fix code for Parser.java

meerian committed:
Fix code for Parser.java

meerian committed:
Add OOP for Ui

meerian committed:
Add OOP for Ui

meerian committed:
Shifted class files to a package

meerian committed:
Create JUnit testing for DateValidator.java and TaskList.java

meerian committed:
Reformat code to fit coding standards

meerian committed:
Fix crash when invalid input was used for done

meerian committed:
Add JDocs documentation to the current class files

meerian committed:
Code now follows the appropriate coding standards

meerian committed:
Add find function

meerian committed:
Merge branch 'branch-Level-9'

meerian committed:
Add javaDocs for find function

shermannws commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,165 @@
+     * which will result in the bot saving the current task list into the file
+     * provided before shutting down.
+     */
+    public void Run() {
```
Since `Run` is a instance method, would you consider adhering to the Java coding standard and use camelCase and name it `run` instead?

shermannws commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * The main program. Keeps on looping until the user inputs "bye",
```
According to the Java coding standard, method header comments' first sentence should start in the form `Returns ...`,`Sends ...`, `Adds ...`. Should you attempt rephrasing the Javadoc for this method?

shermannws commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,165 @@
+
+        while (!Objects.equals(input, "bye")) {
+            switch (input) {
+                case "list":
```
Should there be indentation for the `case` clauses? You might want to refer to the Java coding standard once more.

shermannws commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,105 @@
+     * @param check used to verify what task to create.
+     * @return the relevant task from the strings provided.
+     */
+    public static Task check(String str, String check) throws DukeException {
```
What does the method `check` do? Would you want to specify a little more in the method name as to what this method is trying to check for?

shermannws commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * Interpret the lines in the save file and recreate the correct tasks.
```
According to Java coding standard for Javadoc, there seems to be several occurrences that the Javadoc does not adhere to the standard. According to the Java coding standard, method header comments' first sentence should start in the form `Returns ...`,`Sends ...`, `Adds ...` Should you attempt rephrasing the Javadoc for this method? Consider checking for the rest of the code as well.

shermannws commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,74 @@
+    /**
+     * Represents the file to read/write the TaskList to.
+     */
+    File taskList;
```
Would you want to add an access modifier for your class variable here?

shermannws commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * Sets the task as done.
+     */
+    public void Done() {
```
Similar issue to above, would you consider using camelCase for instance methods?

shermannws reviewed (commented):
All in all, I think your code adhered well to the code quality guidelines in the textbook. Naming of your classes and methods were easily comprehensible and clear too. However, there might be some coding standard and Javadoc standard that you can work on. Good job and keep up the good work!

meerian committed:
Merge pull request #1 from meerian/branch-A-CodingStandard

Code now follows the appropriate coding standards

meerian committed:
Update README.md

meerian committed:
Merge remote-tracking branch 'origin/master'

meerian committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

meerian committed:
Support for gradle running and testing

meerian committed:
Add support for checkstyle

meerian committed:
Merge remote-tracking branch 'origin/A-CheckStyle'

# Conflicts:
#	src/main/java/duke/Main.java
#	src/test/java/DateValidatorTest.java

meerian committed:
Implement basic GUI for Duke

meerian committed:
Add JavaDocs to some functions

meerian committed:
Add JavaDocs to some functions

meerian committed:
Add JavaDocs to some functions

meerian committed:
Merge pull request #2 from meerian/branch-Level-10

Branch level 10

meerian committed:
Add assertions

Adding assertions help to document important assumptions that should hold at various points in the code.

Adding assertions help improves the code quality.

meerian committed:
Improve code quality

Did code refactoring for the overall program, such as removing code duplication.

Code refactoring also helps to improve readability of the code by removing duplicated codes.

meerian committed:
Fix mainClassName for build.gradle

meerian committed:
Merge pull request #3 from meerian/branch-A-Assertions

Add assertions

meerian committed:
Merge pull request #4 from meerian/branch-A-CodeQuality

Improve code quality

meerian committed:
Add app guidance to users.

Added in-app commands to guide new users on how to use DukeBot.

In-app help commands can help new users learn how to use DukeBot in a simple fashion.

Let's abstract all these help commands into a separate window to display a guide for new users instead of having them to type out each command.

meerian committed:
Merge pull request #5 from meerian/branch-C-Help

Add app guidance to users.

meerian committed:
Fix some errors mentioned in the 2103-bot code quality feedback

meerian committed:
Improve GUI by adding backgrounds and new images

meerian committed:
Merge pull request #7 from meerian/branch-A-BetterGui

Improve GUI by adding backgrounds and new images

meerian committed:
Minor code refactoring and addition of 'help' command to provide a guide for users

meerian committed:
Minor code refactoring and addition of 'help' command to provide a guide for users

meerian committed:
Add Ui screenshot and user guide

meerian committed:
Set theme jekyll-theme-cayman

meerian committed:
Set theme jekyll-theme-architect

meerian committed:
Add Ui screenshot and user guide

meerian committed:
Update README.md
